ts/utils/calendar.ts

```diff
@@ -50,3 +51,21 @@ export async function checkCalendarPermission() {
     return false;
   }
 }
+
+/**
+ * This Type has been introduced after this story https://www.pivotaltracker.com/story/show/172079415
+ * to solve the bug on some android devices (mostly the one running MIUI) naming their local calendar
+ * with camel case notation this is a common situation as figured on a related reddit post
+ * https://www.reddit.com/r/Xiaomi/comments/84jgdn/google_calendars_not_syncing_or_even_requesting/
+ * and can be seen on MIUI's github repository
+ * https://github.com/ChameleonOS/miui_framework/blob/master/java/miui/provider/ExtraCalendarContracts.java
+ */
+const localCalendarTitles: ReadonlyArray<string> = [
+  "calendar_displayname_local"
+];
+
+export function convertLocalCalendarName(calendarTitle: string) {
```

Comment:
Since we could add more translations for other calendar titles, could be an improvement define a type that maps the calendar title and its translation key?

What about this?

```typescript
type CalendarTitleTranslation = { [key: string]: TranslationKeys };

const calendarTitleTranslations: CalendarTitleTranslation = {
  calendar_displayname_local: "profile.preferences.calendar.local_calendar"
};

export function convertLocalCalendarName(calendarTitle: string) {
  return fromNullable(calendarTitleTranslations[calendarTitle]).fold(
    calendarTitle,
    s => I18n.t(s)
  );
}
```

Reply:
I was thinking on working with types on `calendarTitleTranslations` such is happening on faq categories but then i obviously thought that this could rise an error on all the cases we didn't track (so also the correct display name).
This is definitely the best solution!

I have also written the test on this function, and added the "birthday calendar" name as listed [here](https://github.com/ChameleonOS/miui_framework/blob/master/java/miui/provider/ExtraCalendarContracts.java)